src/base/static/components/molecules/map-radio-menu.tsx

```diff
@@ -1,102 +1,101 @@
   updateLayerAggregators,
+  layerGroupsSelector,
+  layersSelector,
   LayerGroups,
   Layer,
 } from "../../state/ducks/map-style";
-import { RadioMenuConfig, RadioMenuOption } from "../../state/ducks/map";
+import {
+  RadioMenuConfig,
+  RadioMenuOption,
+  mapRadioMenuConfigSelector,
+} from "../../state/ducks/map";
 import FormLabel from "@material-ui/core/FormLabel";
 import FormControlLabel from "@material-ui/core/FormControlLabel";
 import Radio from "@material-ui/core/Radio";
 import RadioGroup from "@material-ui/core/RadioGroup";
+import MapWidgetWrapper from "./map-widget-wrapper";
 
-import { getReadableColor } from "../../utils/color";
-
-const BACKGROUND_COLOR = "rgba(0, 0, 0, 0.6)";
-const TEXT_COLOR = getReadableColor(BACKGROUND_COLOR);
-
-const styles = (theme: Theme) =>
-  createStyles({
-    label: {
-      color: TEXT_COLOR,
-    },
-  });
+const MapRadioMenu = () => {
+  const mapRadioMenuConfig: RadioMenuConfig = useSelector(
+    mapRadioMenuConfigSelector,
+  );
 
-interface Props extends WithStyles<typeof styles> {
-  radioMenuConfig: RadioMenuConfig;
-  layerGroups: LayerGroups;
-  layers: Layer[];
-  updateLayerAggregators: typeof updateLayerAggregators;
-}
+  const {
+    layerId,
+    layerGroupId,
+    options,
+    defaultSelectedOption,
+    label,
+  }: RadioMenuConfig = mapRadioMenuConfig;
 
-const MapRadioMenu: React.FunctionComponent<Props> = ({
-  radioMenuConfig,
-  layerGroups,
-  layers,
-  updateLayerAggregators,
-  classes,
-}) => {
+  const layerGroups: LayerGroups = useSelector(layerGroupsSelector);
+  const layerGroup = layerGroups.byId[layerGroupId];
+  const layers: Layer[] = useSelector(layersSelector);
+  const dispatch = useDispatch();
   const [selectedOption, setSelectedOption] = React.useState<RadioMenuOption>(
-    radioMenuConfig.options.find(
-      option => option.id === radioMenuConfig.defaultSelectedOption,
-    )!,
+    options.find(option => option.id === defaultSelectedOption)!,
   );
 
   React.useEffect(() => {
-    const layer = layers.find(layer => layer.id === radioMenuConfig.layerId)!;
-    updateLayerAggregators(layer.id, [selectedOption.aggregator!]);
-  }, [radioMenuConfig.layerId, selectedOption]);
+    const layer = layers.find(layer => layer.id === layerId)!;
+    dispatch(updateLayerAggregators(layer.id, [selectedOption.aggregator!]));
+    // Note that `layers` is intentionally left out of the dependency array,
+    // since the combination of `useSelector` above (which always returns a
+    // new reference) and `dispatch` in this effect will cause this effect to
+    // continuously run. Leaving out the dep works because the ids on each layer
+    // do not change, and we can use the initial value of `layers` on each
+    // run of the effect without worrying that we have stale information.
+    // TODO: Leaving out the dependency is probably not correct...
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [layerId, selectedOption, dispatch]);
```

Comment:
This is an interesting problem, but in this case, I think there is a simple solution:

```javascript
  React.useEffect(() => {
    dispatch(updateLayerAggregators(layerId, [selectedOption.aggregator!]));
  }, [layerId, selectedOption, dispatch]);
```

It doesn't look like we need the `layers`, or `layerSelector`, etc.